packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingMethods.js

```diff
@@ -0,0 +1,77 @@
+import React, { Fragment } from 'react';
+import gql from 'graphql-tag';
+import { Form } from 'informed';
+import { useShippingMethods } from '@magento/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingMethods';
+
+import { mergeClasses } from '../../../../classify';
+import Button from '../../../Button';
+import ShippingForm from './shippingForm';
+import defaultClasses from './shippingMethods.css';
+import { ShippingMethodsFragment } from './shippingMethodsFragments';
+import ShippingRadios from './shippingRadios';
+
+const ShippingMethods = props => {
+    const {
+        hasMethods,
+        isShowingForm,
+        selectedShippingFields,
+        selectedShippingMethod,
+        setIsShowingForm,
+        shippingMethods
+    } = useShippingMethods({
+        getShippingMethodsQuery: GET_SHIPPING_METHODS
+    });
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    return (
+        <div className={classes.root}>
+            <p className={classes.message}>
+                For shipping estimates before proceeeding to checkout, please
+                provide the Country, State, and ZIP for the destination of your
+                order.
+            </p>
+            {isShowingForm ? (
+                <Fragment>
+                    <ShippingForm
+                        hasMethods={hasMethods}
+                        selectedShippingFields={selectedShippingFields}
+                    />
+                    {hasMethods ? (
+                        <Fragment>
+                            <h3 className={classes.prompt}>Shipping Methods</h3>
+                            <Form>
+                                <ShippingRadios
+                                    selectedShippingMethod={
+                                        selectedShippingMethod
+                                    }
+                                    shippingMethods={shippingMethods}
+                                />
+                            </Form>
+                        </Fragment>
+                    ) : null}
+                </Fragment>
+            ) : (
+                <Button
+                    classes={{ root_lowPriority: classes.estimateLink }}
+                    priority="low"
+                    type="button"
+                    onClick={() => setIsShowingForm(true)}
```

Comment:
Moved to talon in 23e9cb5, but I'm not sure that solved the problem. I'm a bit opposed to `useCallback` with an empty dependencies list, but let me know if I'm just missing something here.

Reply:
An empty dependencies list just means that the callback function will never change, which is exactly what we want.

```js
const showForm = () => setIsShowingForm(true);
```

As it stands now, `showForm` will get recreated every time through the talon.

Does that make sense?

Reply:
Nope, but I trust you. Added `useCallback` in 2746e6c.

This article from Kent Dodds is why I pushed back; but maybe talons fall into a different render loop that this doesn't apply to.

https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
His problem there is that he's still recreating an anonymous function every render:

```jsx
<button onClick={() => dispense(candy)}>
```

Ours is different because we can say

```jsx
<button onClick={showForm}>
```

which React can tell hasn't changed and therefore the `button` has not received new props so it does not need to re-render.